src/EmergencyMultisig.sol

```diff
@@ -97,144 +89,94 @@ contract EmergencyMultisig is
-        if (newAddresslistLength < multisigSettings.minApprovals) {
-            revert MinApprovalsOutOfBounds({
-                limit: newAddresslistLength,
-                actual: multisigSettings.minApprovals
-            });
-        }
-
-        _removeAddresses(_members);
-
-        emit MembersRemoved({members: _members});
+        return _interfaceId == type(IEmergencyMultisig).interfaceId || _interfaceId == type(IMembership).interfaceId
+            || super.supportsInterface(_interfaceId);
     }
 
     /// @notice Updates the plugin settings.
     /// @param _multisigSettings The new settings.
-    function updateMultisigSettings(
-        MultisigSettings calldata _multisigSettings
-    ) external auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID) {
+    function updateMultisigSettings(MultisigSettings calldata _multisigSettings)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
         _updateMultisigSettings(_multisigSettings);
     }
 
     /// @notice Creates a new multisig proposal.
-    /// @param _metadata The metadata of the proposal.
-    /// @param _actions The actions that will be executed after the proposal passes.
-    /// @param _allowFailureMap A bitmap allowing the proposal to succeed, even if individual actions might revert. If the bit at index `i` is 1, the proposal succeeds even if the `i`th action reverts. A failure map value of 0 requires every action to not revert.
+    /// @param _encryptedPayloadURI The URI where the encrypted contents of the proposal can be found.
+    /// @param _destinationActionsHash The hash of the serialized actions that will be executed after the proposal passes.
+    /// @param _destinationPlugin The address of the plugin to forward the proposal to when it passes.
     /// @param _approveProposal If `true`, the sender will approve the proposal.
-    /// @param _tryExecution If `true`, execution is tried after the vote cast. The call does not revert if early execution is not possible.
-    /// @param _startDate The start date of the proposal.
-    /// @param _endDate The end date of the proposal.
     /// @return proposalId The ID of the proposal.
     function createProposal(
-        bytes calldata _metadata,
-        IDAO.Action[] calldata _actions,
-        uint256 _allowFailureMap,
-        bool _approveProposal,
-        bool _tryExecution,
-        uint64 _startDate,
-        uint64 _endDate
+        bytes calldata _encryptedPayloadURI,
+        bytes32 _destinationActionsHash,
+        OptimisticTokenVotingPlugin _destinationPlugin,
+        bool _approveProposal
     ) external returns (uint256 proposalId) {
-        if (multisigSettings.onlyListed && !isListed(_msgSender())) {
-            revert ProposalCreationForbidden(_msgSender());
+        if (multisigSettings.onlyListed && !multisigSettings.addresslistSource.isListed(msg.sender)) {
```

Comment:
Why have you moved over to msg.sender instead of msgSender()?

Reply:
OpenGSN is deprecated in favour of account abstraction and nobody has been using this feature, ever. There was a discussion with OS and the conclusion was that it was safe to simply move away from `_msgSender()`